### PR TITLE
[KEYCLOAK-14344] Cannot revoke offline access for an app if the app d…

### PR DESCRIPTION
**Pull request by douglaspalmer**
…oesn't require consent

### CODE REVIEW

**Review by vmuzikar — CHANGES_REQUESTED** [review]
It seems like UI tests are failing.
Could you please also add some simple new UI test for this?

**Comment by ssilvert** [comment]
UI tests are passing now.  Build number 936.

@vmuzikar Do you approve to merge?

File: testsuite/integration-arquillian/tests/other/base-ui/src/test/java/org/keycloak/testsuite/ui/account2/ApplicationsTest.java [hunk 1]
```diff
@@ -88,7 +88,7 @@ public void applicationListTest() throws Exception {
         assertFalse(applications.isEmpty());
         Map<String, ApplicationsPage.ClientRepresentation> apps = applications.stream().collect(Collectors.toMap(x -> x.getClientId(), x -> x));
         assertThat(apps.keySet(), containsInAnyOrder("always-display-client", "account-console"));
-        assertClientRep(apps.get("account-console"), "Account Console", false, true, "/realms/test/account/", false);
+        assertClientRep(apps.get("account-console"), "Account Console", false, true, "https://localhost:8543/auth/realms/test/account/", false);
```
Comment (vmuzikar) [inline testsuite/integration-arquillian/tests/other/base-ui/src/test/java/org/keycloak/testsuite/ui/account2/ApplicationsTest.java:91]: Please don't use hardcoded localhost. If we would need to run the tests e.g. with a different host (and we do need that now in some cases to properly test cookies) or without SSL, this won't work.

Something like:
```suggestion
        assertClientRep(apps.get("account-console"), "Account Console", false, true, getAuthServerRoot() + "/realms/test/account/", false);
```

This concerns other lines as well.

**Review by vmuzikar — CHANGES_REQUESTED** [review]

**Review by vmuzikar — DISMISSED** [review]
@douglaspalmer Thanks for the update. If the tests pass, we can merge.

**Comment by ssilvert** [comment]
Kicked off UI tests.  Build number 939.

**Comment by ssilvert** [comment]
Doug kicked off UI tests. Build number 940.

**Comment by ssilvert** [comment]
Build 940 looks good.  ApplicationsTests now passing.

**Review by ssilvert — APPROVED** [review]

**Pull request merged**